client/client-multi/client-objective-c/Kaa/AbstractKaaClient.m

```diff
@@ -0,0 +1,613 @@
+/*
+ * Copyright 2014-2015 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "AbstractKaaClient.h"
+#import "NSMutableArray+Shuffling.h"
+#import "KaaClientPropertiesState.h"
+#import "KeyUtils.h"
+#import "DefaultMetaDataTransport.h"
+#import "DefaultChannelManager.h"
+#import "ProfileManager.h"
+#import "DefaultProfileManager.h"
+#import "DefaultBootstrapManager.h"
+#import "DefaultBootstrapTransport.h"
+#import "DefaultRedirectionTransport.h"
+#import "DefaultLogTransport.h"
+#import "DefaultConfigurationTransport.h"
+#import "DefaultProfileTransport.h"
+#import "DefaultEventTransport.h"
+#import "DefaultNotificationTransport.h"
+#import "DefaultFailoverManager.h"
+#import "DefaultBootstrapDataProcessor.h"
+#import "DefaultOperationDataProcessor.h"
+#import "DefaultBootstrapChannel.h"
+#import "DefaultOperationTcpChannel.h"
+#import "KaaLogging.h"
+#import "KaaExceptions.h"
+
+#define TAG @"AbstractKaaClient >>>"
+#define LONG_POLL_TIMEOUT 60000
+
+@interface AbstractKaaClient ()
+
+@property (nonatomic) volatile BOOL isInitialized;
```

Comment:
I think this flag is not needed not as it a client stores its own state in the lifecycleState property.